internal/stream/processor.go

```diff
@@ -155,3 +160,13 @@ func (pr *processor) pushMessage(ctx context.Context, message *metrov1.ReceivedM
 func (pr *processor) Shutdown() {
 	ants.Release()
 }
+
+func (pr *processor) getSpanContext(message *metrov1.ReceivedMessage) opentracing.SpanContext {
+	spanContext, extractErr := opentracing.GlobalTracer().Extract(opentracing.TextMap, opentracing.TextMapCarrier(message.Message.Attributes))
+	if extractErr != nil {
+		logger.Ctx(pr.ctx).Errorw("failed to get span context from message", "error", extractErr.Error())
+		return nil
+	}
+	delete(message.Message.Attributes, messagebroker.UberTraceID)
```

Comment:
Why are we deleting the traceID from the message? Wouldn't it be re-used at the time of retries?